update.cf

```diff
@@ -43,19 +43,61 @@ body agent control
 
 bundle common update_def
 {
-  vars:
+  classes:
+      # all extra classes are defined true
+      "$(extra_classes)" expression => "any", meta => { "extra" };
+
+      # all override classes are defined true
+      "$(override_classes)" expression => "any", meta => { "override" };
+
+      "have_augments_file" expression => fileexists($(augments_file)), scope => "bundle";
 
+      "augments_are_json" expression => regcmp(".+\.json", $(augments_file)), scope => "bundle";
+      "have_json_augments" and => { "have_augments_file", "augments_are_json" }, scope => "bundle";
+
+      "augments_are_yaml" expression => regcmp(".+\.yaml", $(augments_file)), scope => "bundle";
+      "have_yaml_augments" and => { "have_augments_file", "augments_are_yaml" }, scope => "bundle";
+
+  vars:
       "current_version" string => "3.7.0";
+    any::
+      "augments_file" string => ifelse(fileexists("$(this.promise_dirname)/services/def.yaml"), "$(this.promise_dirname)/services/def.yaml",
+                                      "$(this.promise_dirname)/services/def.json");
+
+      "defvars" slist => variablesmatching("default:def\..*", "defvar");
+
+      "extra_classes" slist => readstringlist("$(this.promise_dirname)/services/extra_classes.txt", "#.*", "\s", 10k, 100k);
+
+    have_augments_file::
+      "augments" data => readjson($(augments_file), 100k), ifvarclass => "augments_are_json";
+      "augments" data => readyaml($(augments_file), 100k), ifvarclass => "augments_are_yaml";
```

Comment:
We cant have yamlstuff until 3.8 since it breaks compatibility with 3.6. I think it's fine to have it there but commented out.